### ISSUE TITLE
How to disable sanity checks

### ISSUE THREAD

**Issue by nwrkbiz** [mesonbuild/meson#12881]
**Describe the bug**
I simply need to disable sanity checks during meson setup

**To Reproduce**
I use meson build setting a proprietary cross compiler using CXX and **I can not** use a toolchain file.

**Expected behavior**
Simply disable the sanity check by e.g. meson setup -Ddisable_sanity_check=true

(also not being able to set a custom linker using CXX_LD causes quite some pain)

**Comment by nwrkbiz** [comment]
Hacked my way around it.

**Comment by xokdvium** [comment]
How did you hack your way around? That would be incredibly helpful. This feature would be incredibly helpful, since not all cross-compiled binaries can run be run on the build machine. E.g. an executable with custom instructions, for which there's no `qemu-user`

**Comment by nwrkbiz** [comment]
> How did you hack your way around? That would be incredibly helpful. This feature would be incredibly helpful, since not all cross-compiled binaries can run be run on the build machine. E.g. an executable with custom instructions, for which there's no `qemu-user`

tl;dr;
I chained together dozend echo commands + the meson setup command that write a cross compile file including 'needs_exe_wrapper = true' before configuring.

**Comment by dcbaker** [comment]
yeah, a side effect of `needs_exe_wrapper = true` is that compiler checks will be skipped if you don't also have an exe_wrapper

**Comment by denizzzka** [comment]
> I chained together dozend echo commands + the meson setup command that write a cross compile file including 'needs_exe_wrapper = true' before configuring.

This works for me:
```ini
[properties]
skip_sanity_check = true
```
